### DISCUSSION TITLE
a/button tag not working inside Listbox.Option

### DISCUSSION THREAD

**Discussion by schultzp2020** [tailwindlabs/headlessui#1668] [Help]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

v1.6.4

**What browser are you using?**

N/A

**Reproduction URL**

[Reproduction URL](https://github.com/schultzp2020/nextjs-template/tree/listbox-issue)
Specifically [VersionSelector](https://github.com/schultzp2020/nextjs-template/blob/listbox-issue/src/components/VersionSelector/VersionSelector.tsx) in `_app.tsx`

**Describe your issue**

The `Listbox` does not navigate correctly when using a `Link` tag from Next.js or a regular anchor tag inside of `Listbox.Option`. After investigating, I have found the href does not fire off every time. Further testing showed if I were to change the Link/a tag to a button tag with an `onClick` attribute, the `onClick` function has the same issues an an href.

**Comment by adamwathan** [comment]
Hey! I think you're using the wrong component here, a Listbox is a form control (like a stylable `<select>`). You would never put a button or link inside of a native `<select>` element.

Check out the `<Menu>` or `<Popover>` components for this sort of thing instead:

https://headlessui.com/react/menu
https://headlessui.com/react/popover

**Reply by schultzp2020** [reply]
Ah, gotcha that makes sense. @adamwathan what's the reason behind using `<Listbox>` instead of lets say `<Menu>` for `<ThemeSelector>` in the TailwindUI Syntax template?

**Reply by schultzp2020** [reply]
@adamwathan I updated `<VersionSelector>` to use `<Menu>` instead of `<Listbox>` and it still does not work 100% of the time. I updated the test repo to reflect my changes.

**Reply by schultzp2020** [reply]
If I set `as={Link}` in `<Menu.Item>` this fixes the issue. However, this seems like it should not be needed.
